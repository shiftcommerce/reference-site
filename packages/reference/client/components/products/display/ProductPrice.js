// Libraries
import { Component } from 'react'

class ProductPrice extends Component {
  renderPrice (variants) {
    const priceList = variants.map((variant) => (variant.price))
    const minPrice = Number(Math.min(...priceList)).toFixed(2)
    const maxPrice = Number(Math.max(...priceList)).toFixed(2)

    if (minPrice !== maxPrice) {
      return (
        <span>
          &pound;{ minPrice } - &pound;{ maxPrice }
        </span>
      )
    } else {
      return (
        <span>
          &pound;{ maxPrice }
        </span>
      )
    }
  }

  render () {
    return (
      <div className='c-product-price'>
        { this.renderPrice(this.props.variants) }
      </div>
    )
  }
}

export default ProductPrice
