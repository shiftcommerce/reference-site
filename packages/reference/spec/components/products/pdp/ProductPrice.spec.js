// Component
import ProductPrice from '../../../../components/products/pdp/ProductPrice'

// Fixtures
import VariantsWithSimilarPrice from '../../../fixtures/variants_with_similar_price.fixture'
import VariantsWithDifferentPrices from '../../../fixtures/variants_with_different_prices.fixture'

test('renders correctly if variants have similar price', () => {
  // Act
  const wrapper = mount(
    <ProductPrice variants={VariantsWithSimilarPrice} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('£10.00')
})

test('renders correctly if variants have different prices', () => {
  // Act
  const wrapper = mount(
    <ProductPrice variants={VariantsWithDifferentPrices} />
  )

  // Assert
  expect(wrapper).toMatchSnapshot()
  expect(wrapper).toIncludeText('£10.00 - £12.00')
})
