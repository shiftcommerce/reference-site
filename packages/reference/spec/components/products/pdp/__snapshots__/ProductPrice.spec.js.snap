// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly if variants have different prices 1`] = `
<ProductPrice
  variants={
    Array [
      Object {
        "price": "10.00",
        "sku": 1,
        "title": "Small",
        "variant": "Size 8",
      },
      Object {
        "price": "11.00",
        "sku": 2,
        "title": "Medium",
        "variant": "Size 10",
      },
      Object {
        "price": "12.00",
        "sku": 3,
        "title": "Large",
        "variant": "Size 12",
      },
    ]
  }
>
  <div
    className="c-product-price"
  >
    <span>
      £
      10.00
       - £
      12.00
    </span>
  </div>
</ProductPrice>
`;

exports[`renders correctly if variants have similar price 1`] = `
<ProductPrice
  variants={
    Array [
      Object {
        "price": "10.00",
        "size": "Size 8",
        "sku": 1,
        "title": "Small",
      },
      Object {
        "price": "10.00",
        "size": "Size 10",
        "sku": 2,
        "title": "Medium",
      },
      Object {
        "price": "10.00",
        "size": "Size 12",
        "sku": 3,
        "title": "Large",
      },
    ]
  }
>
  <div
    className="c-product-price"
  >
    <span>
      £
      10.00
    </span>
  </div>
</ProductPrice>
`;
